teal/editor/src/teal.clj

```diff
@@ -0,0 +1,290 @@
+               :name "support.function.any-method.teal"}
+
+              ;; Self reference
+              {:match #"(?<![^.]\.|:)\b(self)\b"
+               :name "variable.language.self.teal"}
+
+              ;; Type annotations
+              {:begin #":"
+               :end #"(?=\s*[=,;)\]}]|$)"
+               :patterns [{:match #"[a-zA-Z_]\w*"
+                           :name "support.type.teal"}
+                          {:match #"\."
+                           :name "punctuation.accessor.teal"}
+                          {:begin #"<"
+                           :end #">"
+                           :name "type.arguments.teal"
+                           :patterns [{:match #"[a-zA-Z_]\w*"
+                                       :name "support.type.teal"}
+                                      {:match #","
+                                       :name "punctuation.separator.teal"}]}
+                          {:begin #"\{"
+                           :end #"\}"
+                           :name "support.tabletype.teal"
+                           :patterns [{:match #"[a-zA-Z_]\w*"
+                                       :name "support.type.teal"}
+                                      {:match #":"
+                                       :name "punctuation.separator.type.teal"}
+                                      {:match #","
+                                       :name "punctuation.separator.teal"}]}
+                          {:begin #"\bfunction\b"
+                           :end #"(?=\s*[=,;)\]}]|$)"
+                           :begin-captures {0 {:name "keyword.declaration.function.teal"}}
+                           :patterns [{:begin #"\("
+                                       :end #"\)"
+                                       :patterns [{:match #"[a-zA-Z_]\w*"
+                                                   :name "support.type.teal"}
+                                                  {:match #","
+                                                   :name "punctuation.separator.teal"}]}
+                                      {:match #"[a-zA-Z_]\w*"
+                                       :name "support.type.teal"}]}]}
+
+              ;; Variable access (field access)
+              {:match #"(?<=[^.]\.|:)\b([a-zA-Z_]\w*)"
+               :name "variable.other.teal"}
+
+              ;; General identifiers (variables)
+              {:match #"\b([a-zA-Z_]\w*)\b"
+               :name "variable.other.teal"}]})
+
+(defn load-plugin [workspace]
+  (let [language "teal"
+        view-opts {:code {:grammar grammar}}]
+    (g/transact
+      (concat
+        (g/update-property workspace :resource-types update "tl" assoc
+                           :language language
+                           :view-opts view-opts)
+        (g/update-property workspace :resource-types-non-editable update "tl" assoc
+                           :language language
+                           :view-opts view-opts)))))
```

Comment:
The teal resource type is registered earlier as a code transpiler's source node. Here, we only add highlighting and some metadata.